source/common/router/header_parser.cc

```diff
@@ -227,8 +228,18 @@ HeaderParserPtr HeaderParser::configure(
   HeaderParserPtr header_parser(new HeaderParser());
 
   for (const auto& header_value_option : headers_to_add) {
-    const bool append = PROTOBUF_GET_WRAPPED_OR_DEFAULT(header_value_option, append, true);
-    HeaderFormatterPtr header_formatter = parseInternal(header_value_option.header(), append);
+    envoy::config::core::v3::HeaderValueOption::HeaderAppendAction append_action =
+        header_value_option.append_action();
+    // Preserve the old behavior until `append` field is fully deprecated.
+    if (header_value_option.has_append()) {
```

Comment:
If someone sets this but also has set append_action to a non-default value, I would raise an error (please test this).

Reply:
Done. Thanks!